### PR TITLE
added BSN

### PR DESCRIPTION
**Pull request by abbert**
```markdown
| Q             | A
| ------------- | ---
| Bug fix?      | no
| New feature?  | yes
| BC breaks?    | no
| Deprecations? | no
| Tests pass?   | yes
| Fixed tickets | -
```

### CODE REVIEW

**Comment by abbert** [comment]
Sorry for the mess 😕  Created a new repo and a new pull request.

**Comment by abbert** [comment]
https://www.government.nl/topics/identification-documents/contents/the-citizen-service-number

File: src/IsoCodes/Bsn.php [hunk 1]
```diff
@@ -0,0 +1,48 @@
+<?php
+
+namespace IsoCodes;
+
+/**
+ * Class Bsn: In the Netherlands, the citizen service number (BSN) is a unique personal number
```
Comment (Soullivaneuh) [inline src/IsoCodes/Bsn.php]: Just remove the `Class Bsn: `, this is useless, we know that the class is named `Bsn` ! :wink:

File: src/IsoCodes/Bsn.php [hunk 2]
```diff
@@ -0,0 +1,48 @@
...
+ * allocated to everyone registered in the Municipal Personal Records Database.
+ *
+ * @link https://www.government.nl/topics/identification-documents/contents/the-citizen-service-number
+ *
+ * @author  Albert Bakker <hello@abbert.nl>
+ */
+final class Bsn implements IsoCodeInterface
+{
+    /**
+     * BSN validator.
```
Comment (Soullivaneuh) [inline src/IsoCodes/Bsn.php]: Useless comment description too. To be removed IMO.

**Comment by Soullivaneuh** [comment]
Your link is interesting but does not describe how the BSN number is composed. Could you please add both government and wikipedia links?

**Comment by abbert** [comment]
@Soullivaneuh see second link in method comments. This was the only article in English which described how a BSN is composed. The basic idea is as follows:

BSN should be 9 or 8 digits and numeric and can be validated using the following method:

For example BSN = 182269978, then
(9 x 1 = first digit) + (8 x 8 = second) + (7 x 2 = etc) + (6 x 2) + (5 x 6) + (4 x 9) + (3 x 9) + (2 x 7) + (-1 x 8) = 198
198 mod 11 = 0, so 182269978 is a valid BSN

The following article (in Dutch) explains it in detail: https://nl.wikipedia.org/wiki/Elfproef (maybe Google Translate? 😄

**Comment by Soullivaneuh** [comment]
> The following article (in Dutch) explains it in detail: https://nl.wikipedia.org/wiki/Elfproef (maybe Google Translate?

Let's add this link for the moment. :+1:

**Comment by Soullivaneuh** [comment]
@ronanguilloux LGTM for me. :+1:

**Comment by abbert** [comment]
Nice

**Comment by kingkorf** [comment]
Top ouwe 👍

**Pull request merged**